### PR TITLE
[feat]: [CI-9958]: Modify command to get changed files for push trigger

### PR DESCRIPTION
**Pull request by anurag-harness**
## Describe your changes

Current command:  git diff --name-status --diff-filter=MADR lastSuccCommit
Issue with this: If in the pipeline some file in the repo is modified that file also comes in the diff
Updated command: git diff --name-status --diff-filter=MADR lastSuccCommit currentCommit

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54478)

### CODE REVIEW

**Comment by anurag-harness** [comment]
trigger messagecheck

**Comment by anurag-harness** [comment]
trigger uts

**Review by jamie-harness — DISMISSED** [review]

File: product/ci/common/external/util.go [hunk 1]
```diff
@@ -58,9 +58,21 @@ const (
...
 )
 
+func GetChangedFilesPR(ctx context.Context, workspace string, log *zap.SugaredLogger, procWriter io.Writer) ([]types.File, error) {
+	diffFilesCmd := diffFilesCmdPR
+	diffFilesCmd = fmt.Sprintf(diffFilesCmd, gitBin)
+	return getChangedFiles(ctx, workspace, log, procWriter, diffFilesCmd)
+}
+
+func GetChangedFilesPush(ctx context.Context, workspace, lastSuccessfulCommitID, currentCommitID string, log *zap.SugaredLogger, procWriter io.Writer) ([]types.File, error) {
+	diffFilesCmd := diffFilesCmdPush
```
[the remark this answers is not in the document]
Comment (anurag-harness) [reply product/ci/common/external/util.go:68]: Done

**Comment by anurag-harness** [comment]
trigger uts

**Review by jamie-harness — DISMISSED** [review]

**Review by rutvijmehta-harness — DISMISSED** [review]
LGTM

File: product/ci/addon/tasks/runtests.go [hunk 2]
```diff
@@ -359,7 +360,11 @@ func (r *runTestsTask) getTestSelection(ctx context.Context, runner testintellig
 		}
 		log.Infof("Using reference commit: %s", lastSuccessfulCommitID)
 		var errChangedFiles error
-		files, errChangedFiles = getChangedFilesPushTriggerFn(ctx, r.id, lastSuccessfulCommitID, r.log)
+		currentCommitID, shaErr := getSha()
+		if shaErr != nil {
```
[the remark this answers is not in the document]
Comment (anurag-harness) [reply product/ci/addon/tasks/runtests.go]: Removed

**Comment by anurag-harness** [comment]
trigger uts

**Pull request merged**